### PR TITLE
Add ItemStack context to Enchantment#getDamageBonus

### PR DESCRIPTION
**Pull request by diesieben07**
This came up as a request on Discord and the patch is fairly small.

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/extensions/IForgeEnchantment.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.world.entity.MobType;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.enchantment.Enchantment;
+
+public interface IForgeEnchantment
+{
+
```
Comment (sciwhiz12) [inline src/main/java/net/minecraftforge/common/extensions/IForgeEnchantment.java]: Formatting: the empty line should be removed.

File: src/main/java/net/minecraftforge/common/extensions/IForgeEnchantment.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     * @param level the level of the enchantment
+     * @param mobType the mob type being attacked
+     * @param enchantedItem the item used for the attack
+     * @return the damage bonus
+     */
+    default float getDamageBonus(int level, MobType mobType, ItemStack enchantedItem)
+    {
+        return self().getDamageBonus(level, mobType);
+    }
+
```
Comment (sciwhiz12) [inline src/main/java/net/minecraftforge/common/extensions/IForgeEnchantment.java]: Formatting: the empty line should be removed.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Looks fine to me, just needs some tiny formatting fixes.

**Review by sciwhiz12 — APPROVED** [review]

**Comment by Shadows-of-Fire** [comment]
May want to patch an `@Deprecated` onto the non-contextual method.

**Comment by LexManos** [comment]
Simple enough, once the deprecation marker is added feel free  to pull.

**Comment by diesieben07** [comment]
Added the deprecation notice

**Pull request merged**